sotodlib/preprocess/processes.py

```diff
@@ -978,20 +1008,42 @@ def calc_and_save(self, aman, proc_aman):
                                        signal=aman[self.signal])
         bands = np.unique(aman.det_info.wafer.bandpass)
         bands = bands[bands != 'NC']
-        m0 = aman.det_info.wafer.bandpass == bands[0]
-        med0 = np.median(pca_signal.weights[m0,0])
-        med1 = np.median(pca_signal.weights[~m0,0])
-        relcal = np.zeros(aman.dets.count)
-        relcal[m0] = med0/pca_signal.weights[m0,0]
-        relcal[~m0] = med1/pca_signal.weights[~m0,0]
-
-        rc_aman = core.AxisManager(aman.dets, aman.samps,
-                                   core.LabelAxis(name='bandpass',
-                                                  vals=bands))
-        rc_aman.wrap('relcal', relcal, [(0,'dets')])
-        rc_aman.wrap('medians', np.asarray([med0, med1]),
-                     [(0, 'bandpass')])
-        rc_aman.wrap('pca_mode0', pca_signal.modes[0], [(0, 'samps')])
+        if len(bands) == 2:
```

Comment:
I don't understand why there are separate blocks for 1, 2, or other number of bands.  This could all be handled in a single block, which assembles results for each band, and writes them into rc_aman.  Too much duplicated code!

Reply:
I also addressed this.